src/ProjectNeon/Assets/Z_Tests/Features/Effects/InterceptAttackTest.cs

```diff
@@ -0,0 +1,23 @@
+﻿
+using NUnit.Framework;
+
+public sealed class InterceptAttackTests
+{
+    [Test]
+    public void InterceptAttack_ApplyEffect_AttackIsIntercepted()
+    {
+        Member paladin = TestMembers.Create(s => s.With(StatType.MaxHP, 10).With(StatType.Damagability, 1f));
+        Member ally = TestMembers.With(StatType.MaxHP, 10);
+        Member attacker = TestMembers.With(StatType.Attack, 5);
+        Attack attack = new Attack(attacker, ally, 5);
+
+        AllEffects.Apply(new EffectData { EffectType = EffectType.InterceptAttack }, paladin, new MemberAsTarget(ally));
+        BattleEvent.Publish(attack);
```

Comment:
This needs to work correctly, even if there are multiple active subscribers to the Attack event. 

The current design doesn't account for this.